Parsing error: token_mgr_error: Lexical error encountered: <EOF> after : ""

<!--- Provide a general summary of the issue in the Title above -->
On certain syslog messages coming in, extractor fails and showing the terms on the message itself also fails.

The exception is available [here](https://gist.github.com/madchap/1acf24a25695858955dfe5aae4ae1665)

The message looks like this

> pxxx-xx-002 tmm[13116]: 01010045:5: Bandwidth utilization is 83 Mbps, exceeded 75% of Licensed 25 Mbps

The extractor would look like this:

> %{HOST:f5_host} %{NOTSPACE:f5_uw}: %{NOTSPACE}: Bandwidth utilization is %{NUMBER:f5_bandwidth_used;int} Mbps, exceeded %{NUMBER:f5_bandwidth_limit_pct;int}\% of Licensed %{NUMBER:f5_bandwidth_limit_abs;int} Mbps

It works well with grokdebug.herokuapp.com

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
I should be able to see the terms and the message should be parsed.

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
Cannot see the terms nor is the message parsed.

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Locate the message (3 or 4 different types from this source)
2. Click on "show terms" - this triggers the 500 error

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
Trying to parse F5 WAF appliance syslog.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: 2.1.2
* Elasticsearch Version: 2.4.1
* MongoDB Version: 3.2.11
* Operating System: Centos 7
* Browser version: Chrome 55

The problem seems to be an unescaped query parameter for the `/messages/{index}/analyze` endpoint.

```text
java.lang.IllegalArgumentException: java.net.URISyntaxException: Malformed escape pair at index 162: http://localhost:9000/api/messages/graylog2_110/analyze?string=pxxx-xx-002%20tmm[13116]:%2001010045:5:%20Bandwidth%20utilization%20is%2083%20Mbps,%20exceeded%2075%%20of%20Licensed%2025%20Mbps
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.getRequestUri(GrizzlyHttpContainer.java:491)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:366)
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224)
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.URISyntaxException: Malformed escape pair at index 162: http://localhost:9000/api/messages/graylog2_110/analyze?string=pxxx-xx-002%20tmm[13116]:%2001010045:5:%20Bandwidth%20utilization%20is%2083%20Mbps,%20exceeded%2075%%20of%20Licensed%2025%20Mbps
	at java.net.URI$Parser.fail(URI.java:2848)
	at java.net.URI$Parser.scanEscape(URI.java:2978)
	at java.net.URI$Parser.scan(URI.java:3001)
	at java.net.URI$Parser.checkChars(URI.java:3019)
	at java.net.URI$Parser.parseHierarchical(URI.java:3111)
	at java.net.URI$Parser.parse(URI.java:3053)
	at java.net.URI.&lt;init&gt;(URI.java:588)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.getRequestUri(GrizzlyHttpContainer.java:489)
	... 6 more
```

--- a/graylog2-web-interface/src/stores/messages/MessagesStore.js
+++ b/graylog2-web-interface/src/stores/messages/MessagesStore.js
@@ -32,7 +32,7 @@ const MessagesStore = Reflux.createStore({
   },
 
   fieldTerms(index, string) {
-    const url = ApiRoutes.MessagesController.analyze(index, StringUtils.stringify(string)).url;
+    const url = ApiRoutes.MessagesController.analyze(index, encodeURIComponent(StringUtils.stringify(string))).url;
     const promise = fetch('GET', URLUtils.qualifyUrl(url))
       .then(
         response => response.tokens,

